tests/unit/test_memoize.py

```diff
@@ -0,0 +1,33 @@
+import unittest
+import time
+
+from needy.memoize import MemoizeMethod
+
+
+class MemoizeMethodTest(unittest.TestCase):
+    class Foo:
+        def __init__(self, i):
+            self.i = i
+
+        @MemoizeMethod
+        def foo(self, j, k=2):
+            time.sleep(1)
```

Comment:
seems like you could just increment a number instead of using timing, which should be avoided when possible